bot/exts/events/advent_of_code/_cog.py

```diff
@@ -52,13 +60,80 @@ def __init__(self, bot: Bot):
         self.status_task.set_name("AoC Status Countdown")
         self.status_task.add_done_callback(_helpers.background_task_callback)
 
+        self.completer_task.start()
+
+    @tasks.loop(hours=1)
+    async def completer_task(self) -> None:
+        """
+        Every hour check who has completed all 50 stars and give them the completionist role.
+
+        This also removes the role from people who may already have it but have been removed
+        from the leaderboard.
+        """
+        await self.bot.wait_until_guild_available()
+        guild = self.bot.get_guild(Client.guild)
+        completionist_role = guild.get_role(Roles.aoc_completionist)
+
+        aoc_name_to_member_id = {
+            aoc_name: member_id
+            for member_id, aoc_name in await self.account_links.items()
```

Comment:
This will break if a username has more than one account mapped to it. If it doesn't erorr out that can just be handled manually.

Reply:
How does it break? if there is more than one discord id mapped to the same aoc username, then it would just overwrite and result in on of them being present.

Getting into a bad data situation like this is directly blocked in the command anyway, so shouldn't be possible.

Reply:
This should be fine, it can be solved manually